### PR TITLE
add to_epoch method in Event class: Transform Events into an array of Epoch

### PR DESCRIPTION
**Pull request by jonathanduperrier**
add to_epoch method in Event class: Transform Events into an array of Epoch

### CODE REVIEW

File: neo/core/event.py [hunk 1]
```diff
@@ -221,3 +221,25 @@ def as_quantity(self):
...
+        """
+        Transform Events into an array of Epoch 
+        """
+        from neo.core import Epoch, Event, Segment
+
+        print("self.time : " + str(self.times))
+        print("self.label : " + str(self.labels))
+        i = 0
+        duration = []
+        for t in self.times:
```
Comment (JuliaSprenger) [inline neo/core/event.py]: variable t is never used. Maybe it would be better to use 'for i in range(len(self.times)):'?

File: neo/core/event.py [hunk 2]
```diff
@@ -221,3 +221,25 @@ def as_quantity(self):
...
+            if i==0 :
+                duration.append(self.times[i])
+            else :
+                duration.append(self.times[i] - self.times[i-1])
```
Comment (JuliaSprenger) [inline neo/core/event.py]: for a better performace you could use np.diff(self.times) to calculate the durations instead of looping.

**Comment by JuliaSprenger** [comment]
It looks as if you are generating epochs with starting at the times of the events, but having a duration corresponding to the time difference since the previous event. I would have rather expected epochs spanning the time between the events. Is this intended and under which conditions is it useful?
It would also help to add some documentation clarifying this as well as a unittest.
I am also not sure if a function like this should be a method of Events or rather should be available in an extended fashion as a general helper function. @apdavison: Any opinions on that?

**Comment by apdavison** [comment]
@JuliaSprenger yes, I think the indexing is off by one.

We discussed in Lausanne the question of whether to make this a method or a helper function. There are arguments for both. In the end, I think having it as a method gives better discoverability - it is attached to the Event class, which I think makes it easier to find if you already have an Event object, compared to having to search through a potentially long list of helper functions. Of course we could have both, where the helper function calls the method or vice versa.

@jonathanduperrier I think the top priority now is to add some unit tests (in the file `neo/test/coretest/test_event.py`). The example given in #388 would be a good starting point.

**Comment by samuelgarcia** [comment]
This looks also very strange to me.

I add in mind that to_epoch would be:
```python
Epoch(time=event.times[:-1], durations=np.diff(event.times), labels=event.labels[:-1])
```
Here, in thisimplementation, the duration is with the previsous event. I don't get the use case.

**Comment by JuliaSprenger** [comment]
I agree with @samuelgarcia, this should be the default behaviour of the function. However, do you think it would make sense to introduce the option to provide durations in case you don't want to cover the whole recording time with the epochs but only specific time windows after the events?
```python
def to_epoch(durations=None):
    if durations is None:
        times = self.times[:-1]
        durations = np.diff(self.times)[:-1]
        labels = self.labels[:-1]
    else:
        times = self.times
        labels = self.labels
    return Epoch(time=times, durations=durations, labels=labels)
```

File: .travis.yml [hunk 3]
```diff
@@ -13,5 +13,6 @@ install:
 # command to run tests, e.g. python setup.py test
 script:
   nosetests --with-coverage --cover-package=neo
+  assert_array_equal(epoch.magnitude, np.array([7.0, 11.0, 22.0]))
```
Comment (apdavison) [inline .travis.yml]: I suppose you meant to put this line in test_epoch.py?

Comment (apdavison) [reply .travis.yml]: sorry, I meant test_event.py

**Comment by samuelgarcia** [comment]
I propose to postpone thie PR for 0.7.

**Comment by apdavison** [comment]
Since this now has conflicts, I have updated @jonathanduperrier's code and made a new PR: #586

**Pull request closed without merging**